src/lib/WalletInfoCollector.ts

```diff
@@ -82,10 +89,24 @@ export default class WalletInfoCollector {
         }
         onUpdate(walletInfo, await derivedAccountsPromise);
 
-        if (walletType === WalletType.LEGACY) {
+        if (walletType === WalletType.LEGACY && WalletInfoCollector._keyguardClient) {
```

Comment:
Why are you checking for `_keyguardClient` here?
It's always set by the call to `_initializeDependencies`.
If it's only to avoid typescript complaint use `!` instead.